src/Mod/Arch/ArchRoof.py

```diff
@@ -42,241 +52,328 @@ def QT_TRANSLATE_NOOP(ctxt,txt):
+    contains a modified copy of getLineIntersections from that function.
+    '''
+    def getLineIntersections(pt1, pt2, pt3, pt4, infinite1, infinite2):
+        # if pt1:
+            ## first check if we don't already have coincident endpoints ######## we do not want that here ########
+            # if pt1 in [pt3, pt4]:
+                # return [pt1]
+            # elif (pt2 in [pt3, pt4]):
+                # return [pt2]
+        norm1 = pt2.sub(pt1).cross(pt3.sub(pt1))
+        norm2 = pt2.sub(pt4).cross(pt3.sub(pt4))
+
+        if not DraftVecUtils.isNull(norm1):
+            try:
+                norm1.normalize()
+            except Part.OCCError:
+                return []
+
+        if not DraftVecUtils.isNull(norm2):
+            try:
+                norm2.normalize()
+            except Part.OCCError:
+                return []
+
+        if DraftVecUtils.isNull(norm1.cross(norm2)):
+            vec1 = pt2.sub(pt1)
+            vec2 = pt4.sub(pt3)
+            if DraftVecUtils.isNull(vec1) or DraftVecUtils.isNull(vec2):
+                return []  # One of the lines has zero-length
+            try:
+                vec1.normalize()
+                vec2.normalize()
+            except Part.OCCError:
+                return []
+            norm3 = vec1.cross(vec2)
+            denom = norm3.x + norm3.y + norm3.z
+            if not DraftVecUtils.isNull(norm3) and denom != 0:
+                k = ((pt3.z - pt1.z) * (vec2.x - vec2.y)
+                     + (pt3.y - pt1.y) * (vec2.z - vec2.x)
+                     + (pt3.x - pt1.x) * (vec2.y - vec2.z)) / denom
+                vec1.scale(k, k, k)
+                intp = pt1.add(vec1)
+
+                if infinite1 is False and not isPtOnEdge(intp, edge1):
+                    return []
+
+                if infinite2 is False and not isPtOnEdge(intp, edge2):
+                    return []
+
+                return [intp]
+            else:
+                return []  # Lines have same direction
+        else:
+            return []  # Lines aren't on same plane
 
+    pt1, pt2, pt3, pt4 = [edge1.Vertexes[0].Point,
+                          edge1.Vertexes[1].Point,
+                          edge2.Vertexes[0].Point,
+                          edge2.Vertexes[1].Point]
 
-def makeRoof(baseobj=None,facenr=0, angles=[45.,], run = [], idrel = [0,],thickness = [50.,], overhang=[100.,], name="Roof"):
+    return getLineIntersections(pt1, pt2, pt3, pt4, infinite1, infinite2)
 
-    '''makeRoof(baseobj,[facenr],[angle],[name]) : Makes a roof based on
-    a closed wire or an object. You can provide a list of angles, run,
-    idrel, thickness, overhang for each edges in the wire to define the
-    roof shape. The default for angle is 45 and the list is
-    automatically complete to match with number of edges in the wire.
-    If the base object is a solid the roof take the shape.'''
 
+def face_from_points(ptLst):
```

Comment:
Call the argument `points`, instead of `ptLst`. Python is a duck-typing language, meaning that `points` does not necessarily have to be a list type; if the function is written in a general way, it may accept also tuples and dictionaries, or even Numpy arrays, and other data types.

Using a generic name `points` is more readable. Using a type with the name `ptsList`, `ptsDict`, `ptsInt`, etc., is a very static-type way of thinking (C++, Pascal, etc.).

Reply:
One of the issues of the old code was the use of of the variable name 'point' for a list of vectors. This then led to confusion with the 'Point' property. But these are cosmetics that can be addressed in a future update.